osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs

```diff
@@ -143,6 +152,11 @@ private void load()
                     }, scoreDisplayContainer.Add);
                 }
             });
+
+            LoadComponentAsync(new GameplayChatDisplay(room)
+            {
+                Expanded = { Value = true },
+            }, chat => leaderboardFlow.Insert(1, chat));
```

Comment:
I can't currently get my multiplayer connection working and will probably look into that in the mean time, but this made me raise an eyebrow.

Would this scenario still work okay?

```diff
diff --git a/osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs b/osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs
index 3bb76c4a76..fad402a2dd 100644
--- a/osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs
+++ b/osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs
@@ -138,19 +138,22 @@ private void load()
                 Expanded = { Value = true },
             }, l =>
             {
-                foreach (var instance in instances)
-                    leaderboard.AddClock(instance.UserId, instance.GameplayClock);
+                Scheduler.AddDelayed(() =>
+                {
+                    foreach (var instance in instances)
+                        leaderboard.AddClock(instance.UserId, instance.GameplayClock);
 
-                leaderboardFlow.Insert(0, leaderboard);
+                    leaderboardFlow.Insert(0, leaderboard);
 
-                if (leaderboard.TeamScores.Count == 2)
-                {
-                    LoadComponentAsync(new MatchScoreDisplay
+                    if (leaderboard.TeamScores.Count == 2)
                     {
-                        Team1Score = { BindTarget = leaderboard.TeamScores.First().Value },
-                        Team2Score = { BindTarget = leaderboard.TeamScores.Last().Value },
-                    }, scoreDisplayContainer.Add);
-                }
+                        LoadComponentAsync(new MatchScoreDisplay
+                        {
+                            Team1Score = { BindTarget = leaderboard.TeamScores.First().Value },
+                            Team2Score = { BindTarget = leaderboard.TeamScores.Last().Value },
+                        }, scoreDisplayContainer.Add);
+                    }
+                }, 10000);
             });
 
             LoadComponentAsync(new GameplayChatDisplay(room)

```

Reply:
What is this achieving?

Reply:
inserting at index 1 before the main portion is loaded

Reply:
A couple of things:
1. For `FillFlowContainer`, `Insert` sets the layout position, so it's always going to be "index 1" regardless of insertion order.
2. This code is basically copied from `MultiplayerPlayer`, so it's nothing new/untested.
3. I still don't understand how your diff addresses this, or what it's trying to achieve.

Reply:
the diff is just a test to see if things still display right , not a proposal for change

Reply:
Aaaaaaah I see. Then yes, it would work okay, because of point (1) in my response above.